### PR TITLE
Add Redeem Script page

### PR DESCRIPTION
**Pull request by pLabarta**
First draft to solve #1065

- Added new Redeem Script page under Proof-of-Stake
- Updated Using Decrediton "Backup redeem script" warning to state the risks
- Added a similar warning to the PoS/How to Stake page
- Added "9. My VSP went down and I lost the redeem script, is there a way to recover it?" to the FAQ under PoS/VSP.

Added the new page under Proof-of-Stake, it is a much more visible place than the FAQ, but it could be moved to the FAQ if we want to keep the Proof-of-Stake section clean of specific solo/vsp info.

### CODE REVIEW

File: docs/faq/proof-of-stake/voting-service-providers.md [hunk 1]
```diff
@@ -59,6 +59,20 @@ In the unlikely scenario your VSP goes down permanently, you can still vote your
 
 ---
 
+#### 9. My VSP went down and I lost the redeem script, is there a way to recover it?
+
+Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred's block explorer, dcrdata.
+
+1. Look for one of your vote o revoke transactions. This can be done creating a new wallet using your seed phrase, and checking your past tickets.
```
[the remark this answers is not in the document]
Comment (pLabarta) [reply docs/faq/proof-of-stake/voting-service-providers.md:66]: Typo, should be 'vote or revoke'

File: docs/proof-of-stake/redeem-script.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Redeem Script
+
+---
+
+When users sign up with a VSP, both the user and the VSP jointly create a 1-of-2 multisignature script. This script is known as the redeem script.
```
[the remark this answers is not in the document]
Comment (pLabarta) [reply docs/proof-of-stake/redeem-script.md:5]: Agreed, added the link and spelled out the acronym

File: docs/proof-of-stake/redeem-script.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+The redeem script involves both a public key from the user and a public key from the VSP so that both can vote and revoke the ticket. When the user buys a new ticket, the voting rights are given to a hash of that redeem script.
+
+---
+
+## Why users must backup their scripts
+
+At first, when the user just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.
+
+The VSP handles voting and revoking for the user, but if it goes offline is up to the user to vote or revoke, which means the wallet needs to know the redeem script.
```
[the remark this answers is not in the document]
Comment (pLabarta) [reply docs/proof-of-stake/redeem-script.md:15]: Maybe "...if it goes offline, it is up to the user to vote using the remaining active tickets or revoke those that have expired. In order to do this, the user's wallet needs to know the redeem script."

File: docs/proof-of-stake/redeem-script.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## What happens if the script is lost?
+
+User who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets. This means their funds will remain locked.
+
+---
+
+## Recovery methods
+
+Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred’s block explorer, [dcrdata](https://explorer.dcrdata.org).
```
[the remark this answers is not in the document]
Comment (pLabarta) [reply docs/proof-of-stake/redeem-script.md:29]: I've left the instructions on the Redeem Script page and added a brief explanation:

> When a ticket votes or is revoked, that redeem script is revealed as a part of the process. Redeem scripts that have been used at least once to vote or revoke a ticket can be [recovered using Decred's block explorer](/proof-of-stake/redeem-script/#recovery-methods).

Almost forgot about the boxes

File: docs/faq/proof-of-stake/voting-service-providers.md [hunk 5]
```diff
@@ -59,6 +59,20 @@ In the unlikely scenario your VSP goes down permanently, you can still vote your
...
+
+2. Open the transaction using [dcrdata](https://dcrdata.org) and look into the decoded JSON to find the `scriptSig`. It can be found inside the `vin` key.
+
+3.  The `asm` value should be a long string separated by a white space. Grab the second part of the `scriptSig` `asm`, this is your redeem script.
```
[the remark this answers is not in the document]
Comment (pLabarta) [reply docs/faq/proof-of-stake/voting-service-providers.md]: Reads much better!

File: docs/proof-of-stake/redeem-script.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (s-ben) [reply docs/proof-of-stake/redeem-script.md]: Your suggested wording is clearer. Let's use it.

File: docs/proof-of-stake/redeem-script.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (s-ben) [reply docs/proof-of-stake/redeem-script.md]: Lgtm.

**Comment by s-ben** [comment]
@pLabarta lgtm. If you implement your suggested changes I'm ready to approve. 

I'm hesitant to merge without a technical review, as redeem scripts aren't my domain. Perhaps @jholdstock can take a quick pass?

File: docs/faq/proof-of-stake/voting-service-providers.md [hunk 6]
```diff
@@ -59,6 +59,12 @@ In the unlikely scenario your VSP goes down permanently, you can still vote your
 
 ---
 
+#### 9. My VSP went down and I lost the redeem script, is there a way to recover it?
+
+When a ticket votes or is revoked, that redeem script is revealed as a part of the process. Redeem scripts that have been used at least once to vote or revoke a ticket can be [recovered using Decred's block explorer](/proof-of-stake/redeem-script/#recovery-methods).
```
Comment (jholdstock) [inline docs/faq/proof-of-stake/voting-service-providers.md:64]: Internal links need to be relative paths to `.md` files in order to benefit from of the link validation built into mkdocs.

In this case we want to use `../../proof-of-stake/redeem-script.md#recovery-methods` (also note removal of the final forward slash

File: docs/proof-of-stake/how-to-stake.md [hunk 7]
```diff
@@ -23,6 +23,11 @@ A list of Voting Service Providers (VSPs) and statistics is maintained on the
 
 Using a VSP **does not give the VSP access to your funds**. All you are doing is granting voting rights to the VSP.
 
+!!! warning "Warning"
```
Comment (jholdstock) [inline docs/proof-of-stake/how-to-stake.md:26]: Seeing as this warning is duplicated on two pages (and may be used on other pages in future) I recommend adding this warning to `mkdocs.yml`, similar to `seedWarning1`, `seedWarning2` etc. See `decrediton-setup.md` for an example usage.

File: docs/proof-of-stake/redeem-script.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# <img class="dcr-icon" src="/img/dcr-icons/QuestionTicket.svg" /> Redeem Script
+
+---
+
+When users sign up with a [Voting Service Provider](/glossary/#voting-service-provider) (VSP), both the user and the VSP jointly create a 1-of-2 multisignature script. This script is known as the redeem script.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:5]: Link to `.md` file

File: docs/proof-of-stake/redeem-script.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+The redeem script involves both a public key from the user and a public key from the VSP so that both can vote and revoke the ticket. When the user buys a new ticket, the voting rights are given to a hash of that redeem script.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:7]: Two things here:

- A ticket can only be voted **or** revoked, not both.
- I think these two sentences actually make more sense if the order is reversed.

Suggest changing to

> When a VSP user buys a new ticket, the voting rights are given to a hash of the redeem script. The redeem script involves both a public key from the user and a public key from the VSP, so either are able to vote or revoke the purchased tickets.

File: docs/proof-of-stake/redeem-script.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+---
+
+## Why users must backup their scripts
+
+At first, when the user just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:13]: At first, when the user has just signs up, the redeem script is not known by anyone except the user and the VSP. When the ticket votes or is revoked, that redeem script is revealed as a part of the process.
```suggestion
At first, when a user has just signed up to a VSP, the redeem script is not known by anyone except the user and the VSP. When a ticket votes or is revoked, that redeem script is revealed on-chain as a part of the process.
```

File: docs/proof-of-stake/redeem-script.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+The VSP handles voting and revoking for the user, but if it goes offline it is up to the user to vote or revoke the tickets, which means the wallet needs to know the redeem script.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:15]:
```suggestion
The VSP handles voting and revoking for the user, but if it goes offline it is up to the user to vote or revoke the tickets, which means the users wallet also needs to know the redeem script.
```

File: docs/proof-of-stake/redeem-script.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+A user who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets. This means their funds will remain locked.
+
+---
+
+## Recovery methods
+
+Only redeem scripts that have been used at least once to vote or revoke a ticket can be recovered. If this is your case, you can find it using Decred's block explorer, [dcrdata](https://dcrdata.org).
+
+1. Look for one of your vote or revoke transactions. This can be done creating a new wallet using your seed phrase, and checking your past tickets.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:31]: Numbered bullets should all use `1.` so they are automatically incremented. This removes the need to update all lines if bullets are added/removes in future.

File: docs/proof-of-stake/how-to-stake.md (hunk 7 above)
Comment (jholdstock) [reply docs/proof-of-stake/how-to-stake.md:26]: We should also use this new warning to replace the one in `using-decrediton.md`

**Comment by pLabarta** [comment]
Sorry folks. Tough week. Just updated the following:

- reworded the redeem-script page to state under which conditions funds may be end up locked
- used relative path in internal links (thanks jholdstock)
- applied several suggested changes

Issues:

- Relative paths break when using them form different sources, like when including a warning. Should we use absolute paths there?

File: docs/proof-of-stake/how-to-stake.md (hunk 7 above)
Comment (pLabarta) [reply docs/proof-of-stake/how-to-stake.md:26]: Added the warning as scriptWarning1 and scriptWarning2

File: docs/proof-of-stake/redeem-script.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+When a VSP user buys a new ticket, the voting rights are given to a hash of the redeem script. The redeem script involves both a public key from the user and a public key from the VSP, so either are able to vote or revoke the purchased tickets.
+
+---
+
+## Why users must backup their scripts
+
+At first, when a user has just signed up to a VSP, the redeem script is not known by anyone except the user and the VSP. When a ticket votes or is revoked, that redeem script is revealed on-chain as a part of the process.
+
+The VSP handles voting and revoking for the user, but if it goes offline, it is up to the user to vote using the remaining active tickets or revoke those that have expired. In order to do this, the users wallet also needs to know the redeem script.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:15]:
```suggestion
The VSP handles voting and revoking for the user, but if it goes offline, it is up to the user to vote the remaining active tickets or revoke those that have expired. In order to do this, the users wallet also needs to know the redeem script.
```

File: docs/proof-of-stake/redeem-script.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+When registering with a VSP, the user wallet will automatically store a copy of the redeem script, but currently, users can't reconstruct these scripts from their own seed. This is why they must [backup their redeem scripts](../wallets/decrediton/using-decrediton.md#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:17]:
```suggestion
When registering with a VSP, the users wallet will automatically store a copy of the redeem script, but currently, users can't reconstruct these scripts from their own seed. This is why they must [backup their redeem scripts](../wallets/decrediton/using-decrediton.md#backup-redeem-script) to ensure they don't end up with locked funds due to the VSP going offline.
```

File: docs/proof-of-stake/redeem-script.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+---
+
+## What happens if the script is lost?
+
+A user who purchased a ticket through a VSP that is no longer available and lost their redeem script won't be able to vote or revoke their tickets.
+
+Users risk having their funds locked if the following conditions are met:
+
+- The user has not backed up their redeem script as the documentation and software instructs
```
Comment (jholdstock) [inline docs/proof-of-stake/redeem-script.md:27]:
```suggestion
- The user has not backed up their redeem script
```

File: mkdocs.yml [hunk 16]
```diff
@@ -17,6 +17,10 @@ extra:
     "Every seed phrase is also associated with a 64 character seed hex. The seed hex functions the same way as the seed phrase, so it is also important to keep your seed hex secure."
   seedWarning4:
     "**DO NOT, UNDER ANY CIRCUMSTANCES, GIVE YOUR SEED OR THE ASSOCIATED HEX KEY TO ANYONE! NOT EVEN DECRED DEVELOPERS!**"
+  scriptWarning1:
+    "Only you and your VSP know the [redeem script](/proof-of-stake/redeem-script). You must backup the script to ensure you don't end up with locked funds. If your VSP goes down, you'll need this script to vote or revoke tickets."
```
Comment (jholdstock) [inline mkdocs.yml:21]:
```suggestion
    "When you register with a VSP, it is essential to backup your "[redeem script](/proof-of-stake/redeem-script)". Only you and your VSP know the redeem script, and it must be backed up to ensure you don't end up with locked funds. If your VSP goes down, you'll need the script to vote or revoke tickets."
```

**Comment by pLabarta** [comment]
All good now!

**Pull request merged**